# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerParent.h
# Commit: 2cf5cad90099
# Full Hash: 2cf5cad900992236d596bfb329099742ff15f689
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-06-20 04:08:16
# Regressor Bug: 1728331
# File Overlap Count: 1
# Description:
#   Bug 1728331 - Part 4: Make ContentParent KeepAlives explicit with RAII references, r=smaug,dom-worker-reviewers,asuth
#   
#   This is a fairly significant patch, however it would be difficult to break it
#   down into smaller patches:
#   
# ==============================================================================

diff -r d920c2d72d00 -r 2cf5cad90099 dom/workers/remoteworkers/RemoteWorkerParent.h
--- a/dom/workers/remoteworkers/RemoteWorkerParent.h	Wed Jun 19 20:14:49 2024 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerParent.h	Wed Jun 19 20:14:50 2024 +0000
@@ -8,6 +8,7 @@
 #define mozilla_dom_RemoteWorkerParent_h
 
 #include "mozilla/dom/PRemoteWorkerParent.h"
+#include "mozilla/dom/UniqueContentParentKeepAlive.h"
 
 namespace mozilla::dom {
 
@@ -25,9 +26,8 @@
  public:
   NS_INLINE_DECL_REFCOUNTING(RemoteWorkerParent, override);
 
-  RemoteWorkerParent();
-
-  void Initialize(bool aAlreadyRegistered = false);
+  explicit RemoteWorkerParent(
+      UniqueThreadsafeContentParentKeepAlive aKeepAlive);
 
   void SetController(RemoteWorkerController* aController);
 
@@ -58,6 +58,7 @@
 
   bool mDeleteSent = false;
   RefPtr<RemoteWorkerController> mController;
+  UniqueThreadsafeContentParentKeepAlive mContentParentKeepAlive;
 };
 
 }  // namespace mozilla::dom